yogstation/code/modules/mining/lavaland/doppelmirror.dm

```diff
@@ -0,0 +1,312 @@
+#define COOLDOWN_RECALL 5 SECONDS
+#define COOLDOWN_MIRRORSEARCH 10 SECONDS
+
+/obj/item/dopmirror
+	name = "ominous mirror"
+	desc = "What do you see looking back at you?"
+	icon = 'yogstation/icons/obj/lavaland/artefacts.dmi'
+	icon_state = "mirrornormal"
+	actions_types = list(/datum/action/item_action/mirrorrecall, /datum/action/item_action/rerollmirror)
+	COOLDOWN_DECLARE(next_search)
+	COOLDOWN_DECLARE(next_recall)
+	var/mob/living/carbon/original = null
+	var/mob/living/simple_animal/hostile/double/reflected = null
+
+/obj/item/dopmirror/pickup(mob/user)
+	..()
+	original = user
+
+/obj/item/dopmirror/attack_self(mob/living/user)
+	if(reflected)
+		return
+	if(!(GLOB.ghost_role_flags & GHOSTROLE_STATION_SENTIENCE))
+		to_chat(user, span_notice("Anomalous otherworldly energies keep the mirror from reflecting anything!"))
+		return
+	if(!COOLDOWN_FINISHED(src, next_search))
+		return
+	COOLDOWN_START(src, next_search, COOLDOWN_MIRRORSEARCH)
+
+	to_chat(user, "You peer into the mirror...")
+	var/list/mob/dead/observer/candidates = pollGhostCandidates("Do you want to play as the living reflection in service of [user.real_name]?", ROLE_PAI, null, FALSE, 100, POLL_IGNORE_POSSESSED_BLADE)
+	if(LAZYLEN(candidates))
+		var/mob/dead/observer/C = pick(candidates)
+		var/mob/living/simple_animal/hostile/double/S = new(src)
+		reflected = S
+		S.ckey = C.ckey
+		S.fully_replace_character_name(null, "living reflection")
+		S.copy_languages(user, LANGUAGE_MASTER)	
+		S.update_atom_languages()
+		S.mirror = src
+		S.faction = user.faction
+		to_chat(S, span_warning("<b>As a resident of the mirror, your topmost priority is the safety of your home.</b>"))
+		to_chat(S, span_warning("<b>Physically defending the mirror without a host is impossible. Ensure the mirror is always in someone's hands so you can act.</b>"))
+		to_chat(S, span_warning("<b>You are moderately strong and are able to deal stronger blows to typical lavaland fauna.</b>"))
+		to_chat(S, span_warning("<b>You are fragile but cannot truly be destroyed until the mirror is. You are able to recover inside the mirror, and excessive damage will seal\
+		 you inside of it for 20 seconds.</b>"))
+		to_chat(S, span_warning("<b>Additionally, you possess the ability to swap places with the current host while outside the mirror.</b>"))
+		grant_all_languages(FALSE, FALSE, TRUE)
+		to_chat(user, "... and you see your reflection trying to leave!")
+		playsound(src, 'sound/effects/glassbr2.ogg', 75)
+		src.update_icon()
+	else
+		to_chat(user, "... and your reflection stares back at you. Try again later.")
+
+/obj/item/dopmirror/Destroy()
+	to_chat(reflected, span_userdanger("Your world shatters."))
+	reflected.jumpback.Remove(reflected)
+	reflected.appear.Remove(reflected)
+	reflected.swap.Remove(reflected)	
```

Comment:
put these instead on the destroy proc of the mob itself
and make sure to qdel them after removing